File: app/src/main/java/br/com/pedrosilva/tecnonutri/util/AppUtil.java
package br.com.pedrosilva.tecnonutri.util;

import android.util.TypedValue;

import java.text.DateFormat;
import java.text.DecimalFormat;
import java.text.NumberFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

import br.com.pedrosilva.tecnonutri.R;

import static br.com.pedrosilva.tecnonutri.util.ContextHelper.getApplicationContext;

/**
 * Created by psilva on 3/17/17.
 */

public abstract class AppUtil {
    private static final String PATTERN_NUMBER = "#.###";
    private static final String PATTERN_KCAL = "#.#";
    private static final String PATTERN_WEIGHT = "#.#";
    private static final String PATTERN_MONEY = "#0.00";

    public static <T> T get(Object element) {
        return (T) element;
    }

    public static String formatDate(Date date) {
        return formatDate(date, getApplicationContext().getString(R.string.date_format));
    }

    private static String formatDate(Date date, String pattern) {
        DateFormat dateTimeFormat = new SimpleDateFormat(pattern, Locale.getDefault());
        return dateTimeFormat.format(date);
    }

    private static String formatDecimal(Number number, String pattern) {
        final DecimalFormat decimalFormat = AppUtil.get(NumberFormat.getNumberInstance(Locale.getDefault()));
        decimalFormat.applyPattern(pattern);
        return decimalFormat.format(number);
    }

    public static String formatDecimal(Number number) {
        return formatDecimal(number, AppUtil.PATTERN_NUMBER);
    }

    public static String formatKcal(Number number) {
        return formatDecimal(number, AppUtil.PATTERN_WEIGHT);
    }

    public static String formatWeight(Number number) {
        return formatDecimal(number, AppUtil.PATTERN_WEIGHT);
    }

    public static int dp2Px(int dp) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 500, ContextHelper.getApplicationContext().getResources().getDisplayMetrics());
    }
}

File: app/src/main/kotlin/br/com/pedrosilva/tecnonutri/util/AppUtil.kt
package br.com.pedrosilva.tecnonutri.util

import java.text.DecimalFormat
import java.text.NumberFormat
import java.util.Locale

object AppUtil {

    private const val PATTERN_NUMBER = "#.###"
    private const val PATTERN_KCAL = "#.#"
    private const val PATTERN_WEIGHT = "#.#"
    private const val PATTERN_MONEY = "#0.00"

    @Suppress("UNCHECKED_CAST")
    operator fun <T> get(element: Any): T = element as T

    private fun formatDecimal(number: Number, pattern: String): String {
        val decimalFormat =
            get<DecimalFormat>(NumberFormat.getNumberInstance(Locale.getDefault()))
        decimalFormat.applyPattern(pattern)
        return decimalFormat.format(number)
    }

    fun formatDecimal(number: Number): String =
        formatDecimal(number, PATTERN_NUMBER)

    fun formatKcal(number: Number): String =
        formatDecimal(number, PATTERN_KCAL)

    fun formatWeight(number: Number): String =
        formatDecimal(number, PATTERN_WEIGHT)
}